testify/test_runner.py

```diff
@@ -252,22 +246,14 @@ def list_suites(self):
 
     def get_tests_for_suite(self, selected_suite_name):
         """Gets the test list for the suite"""
-        test_list = []
         for test_instance in self.discover():
             for test_method in test_instance.runnable_test_methods():
                 if not selected_suite_name or TestCase.in_suite(test_method, selected_suite_name):
-                    test_list.append(test_method)
-        return test_list
+                    yield test_method
 
     def list_tests(self, selected_suite_name=None):
         """Lists all tests, optionally scoped to a single suite."""
-        test_list = self.get_tests_for_suite(selected_suite_name)
-        test_method_names = sorted([
-            self.get_test_method_name(test) for test in test_list
-        ])
-        for test_method_name in test_method_names:
-            print(test_method_name)
-
-        return test_list
+        for test in self.get_tests_for_suite(selected_suite_name):
+            print(self.get_test_method_name(test))
```

Comment:
I've confirmed that discovery on yelp-main is deterministic and sorted (matches `LANG=C sort`).
Shall I add some kind of assertion?